runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/middleware/AwsSigningMiddleware.kt

```diff
@@ -190,3 +212,9 @@ private fun HttpRequestBuilder.update(signedRequest: HttpRequest) {
         }
     }
 }
+
+/**
+ * @return a boolean representing if this HttpBody is eligible to send via aws-chunked content encoding
+ */
+private val HttpBody.isEligibleForAwsChunkedStreaming: Boolean
+    get() = !(this as HttpBody.Streaming).isReplayable || (this.contentLength ?: 0) > AwsSigningMiddleware.AWS_CHUNKED_THRESHOLD
```

Comment:
nit: This `as` is only safe because we use it after checking it's `HttpBody.Streaming` on line 152, we could just remove the check on line 152 and move the logic completely in here:

```kotlin
private val HttpBody.isEligibleForAwsChunkedStreaming: Boolean
    get() = (this is HttpBody.Streaming) && (!isReplayable) || (contentLength ?: 0) > AwsSigningMiddleware.AWS_CHUNKED_THRESHOLD)
```

I also think this should probably only apply to streaming bodies, for large in-memory bodies we should just calculate the hash from the payload (the updated logic here would handle that)